src/main/java/com/rtb/projectmanagementtool/project/ProjectData.java

```diff
@@ -41,32 +46,51 @@ public ProjectData(String name, String description, long creatorId) {
    */
   public ProjectData(Entity entity) {
     this.id = (Long) entity.getKey().getId();
+    this.creatorId = (Long) entity.getProperty(PROPERTY_CREATOR);
     this.name = (String) entity.getProperty(PROPERTY_NAME);
     this.description = (String) entity.getProperty(PROPERTY_DESCRIPTION);
 
-    // Tasks
-    Object entityProperty;
-    if ((entityProperty = entity.getProperty(PROPERTY_TASKS)) != null) {
-      this.tasks = (ArrayList<Long>) entityProperty;
-    } else {
-      this.tasks = new ArrayList<Long>();
+    // Instantiate containers
+    this.members = new HashSet<Long>();
+    this.admins = new HashSet<Long>();
+
+    // Update containers with entity properties
+    parseEntityHashSets(entity, PROPERTY_ADMINS);
+    parseEntityHashSets(entity, PROPERTY_MEMBERS);
+  }
+
+  /**
+   * Used by ProjectData(Entity) to parse entity properties into respective HashSets of this class
+   *
+   * @param entity the entity
+   * @param propertyName the name of entity property
+   */
+  private void parseEntityHashSets(Entity entity, String propertyName) {
```

Comment:
Is this still needed?  I believe you can cast directly from a datastore array to a java arraylist